scraper/src/libretexts2zim/css.py

```diff
@@ -0,0 +1,140 @@
+        target_parent = Path(
+            *[
+                parent.name
+                for parent in reversed(original_path.parents)
+                if parent.name and parent.name != ".."
+            ]
+        )
+
+        index = 0
+        while True:
+            relative_path = (
+                target_parent
+                / f"{original_path.stem}{'_' + str(index) if index else ''}"
+                f"{original_path.suffix}"
+            )
+            if relative_path not in self.used_paths:
+                break
+            index += 1
+
+        self.used_paths.append(relative_path)
+        target_path = self.css_assets_root_path / relative_path
+        self.css_assets[original_url] = target_path
+        return target_path.relative_to(self.css_target_path)
+
+    def _process_node(self, css_original_url: str, node: ast.Node):
+        if isinstance(
+            node,
+            ast.QualifiedRule
+            | ast.SquareBracketsBlock
+            | ast.ParenthesesBlock
+            | ast.CurlyBracketsBlock,
+        ):
+            self._process_list(
+                css_original_url,
+                node.content,  # pyright: ignore[reportUnknownArgumentType, reportUnknownMemberType]
+            )
+        elif isinstance(node, ast.FunctionBlock):
+            if node.lower_name == "url":  # pyright: ignore[reportUnknownMemberType]
+                url_node: ast.Node = node.arguments[0]  # pyright: ignore
+                relative_css_path = self._process_url(
+                    css_original_url,
+                    url_node.value,  # pyright: ignore
+                )
+                if not relative_css_path:
+                    return
+                url_node.value = str(relative_css_path)  # pyright: ignore
+                url_node.representation = (  # pyright: ignore
+                    f'"{serialize_url(str(relative_css_path))}"'
+                )
+
+            else:
+                self._process_list(
+                    css_original_url,
+                    node.arguments,  # pyright: ignore
+                )
+        elif isinstance(node, ast.AtRule):
+            self._process_list(
+                css_original_url,
+                node.prelude,  # pyright: ignore
+            )
+            self._process_list(
+                css_original_url,
+                node.content,  # pyright: ignore
+            )
+        elif isinstance(node, ast.Declaration):
+            self._process_list(
+                css_original_url,
+                node.value,  # pyright: ignore
+            )
+        elif isinstance(node, ast.URLToken):
+            relative_css_path = self._process_url(
+                css_original_url,
+                node.value,  # pyright: ignore
+            )
+            if not relative_css_path:
+                return
+            node.value = str(relative_css_path)
+            node.representation = f"url({serialize_url(str(relative_css_path))})"
```

Comment:
representations usually are capitalized (because usually those are types) ; not sure if appropriate though

Reply:
`node` and `node.representation` comes from tinycss2, not something we can choose. Or do I miss your point?